### ISSUE TITLE
Possible NullReferenceException when using Nested Classes in Model

### ISSUE THREAD

**Issue by jayharris** [alexanderar/Mvc.CascadeDropDown#29]
**tl;dr**
When the `expression` argument is for a property on a sub-model, such as `Model.Address.SelectedCity`, and the model is instantiated but the submodel is null, rendering will fail with a NullReferenceException.

**Detail:**
When evaluating the current property value during render so that it can be pre-selected in the drop-down, rendering can fail if the property is on an uninstantiated sub-class / sub-model.

If the control's `expression` argument is for a property that is directly on the model such as `Model.SelectedCity`, and the model is `null`, the control handles the `null` model instead of throwing. However, if the model is instantiated, but the control's `expression` argument is for a property on a sub-model such as `Model.CurrentAddress.SelectedCity`, and the submodel is `null`, evaluating the expression throws a NullReferenceException and rendering fails.

```
namespace Mvc.CascadeDropDown.Test.Models
{
    using System.ComponentModel.DataAnnotations;
    using System.Web.Mvc;

    public class CascadingDropdownsModel
    {
        public IList<SelectListItem> Countries { get; set; }

        public AddressModel CurrentAddress { get; set; }
    }

    public class AddressModel
    {
        [Required]
        public string SelectedCountry { get; set; }

        [Required]
        public string SelectedCity { get; set; }

        [Required]
        public int? SelectedStreet { get; set; }
    }
}
```

```
        public ActionResult Index()
        {
            var model = new CascadingDropdownsModel
            {
                Countries =
                    new List<SelectListItem>
                    {
                        new SelectListItem
                        {
                            Text = "US",
                            Value = "US"
                        },
                        new SelectListItem
                        {
                            Text = "UK",
                            Value = "UK"
                        },
                    },
                CurrentAddress = null
            };
            return View(model);
        }
```

```
        <div class="form-group">
            @Html.CascadingDropDownListFor(expression: m => m.CurrentAddress.SelectedCity,
                                triggeredByProperty: m => m.CurrentAddress.SelectedCountry,
                                url: Url.Action("GetCities", "Home"),
                                ajaxActionParamName: "country",
                                disabledWhenParentNotSelected: true,
                                htmlAttributes: new { @class = "form-control" },
                                options: new CascadeDropDownOptions
                                {
                                    BeforeSend = "beforeSend",
                                    HttpMethod = "post",
                                    OnCompleteGetData = "onComplete",
                                    OnFailureGetData = "onFailure",
                                    OnSuccessGetData = "onSuccess"
                                })
            @Html.ValidationMessageFor(m => m.CurrentAddress.SelectedCity)
        </div>
```

The exception is thrown within `GetPropStringValue` when the control is attempting to get the current value of the property.

https://github.com/alexanderar/Mvc.CascadeDropDown/blob/5641ced7fccd8975bc124c11e044978cf72d1e8a/Mvc.CascadeDropDown/DropDownListExtensions.cs#L539

**Expected Behavior:**
The control handles the NullReferenceException when evaluating the `expression` argument, returning an empty string similar to when the model is `null`, and allowing the page to fully render.

**Comment by alexanderar** [comment]
Hi Jay,
Thanks for your posting this issue, I will check it out.
You are also welcome to submit a PR with a fix.

**Comment by alexanderar** [comment]
@jayharris 
I fixed the issue and released a [new 2.0.2 version](https://www.nuget.org/packages/Mvc.CascadeDropDown/2.0.2).

I am waiting for your feedback on this issue to close ticket.
